src/ray/gcs/gcs_server/gcs_actor_manager.h

```diff
@@ -441,6 +449,9 @@ class GcsActorManager : public rpc::ActorInfoHandler {
   /// This method MUST BE IDEMPOTENT because it can be called multiple times during
   /// actor destroy process.
   std::function<void(const ActorID &)> destroy_owned_placement_group_if_needed_;
+  /// A callback to get the namespace an actor belongs to based on its job id. This is
+  /// necessary for actor creation.
+  std::function<std::string(const JobID &)> get_namespace_;
```

Comment:
TODO We need to populate this mapping on initialization in case of a GCS restart.